worker/registry.ts

```diff
@@ -77,22 +81,45 @@ export async function handleRegistryRequest(url: URL): Promise<Response> {
   }
   const remoteUrl = getBackingURL(module, version, path);
   const resp2 = await fetchSource(remoteUrl);
-
-  // JSX and TSX content type fix
+  let respText = await resp2.text();
   if (
-    remoteUrl.endsWith(".jsx") &&
-    !resp2.headers.get("content-type")?.includes("javascript")
+    headers.get("accept")?.includes("application/typescript") ||
+    headers.get("user-agent")?.includes("Deno/")
   ) {
+    // JSX and TSX content type fix
+    if (
+      remoteUrl.endsWith(".jsx") &&
+      !resp2.headers.get("content-type")?.includes("javascript")
+    ) {
+      resp2.headers.set("content-type", "application/javascript");
+    } else if (
+      remoteUrl.endsWith(".tsx") &&
+      !resp2.headers.get("content-type")?.includes("typescript")
+    ) {
+      resp2.headers.set("content-type", "application/typescript");
+    }
+  } else if (headers.get("accept")?.includes("javascript")) {
```

Comment:
Maybe we should also include the case of `Accept: */*`?

Reply:
I decided against this because if you aren't requesting anything specific you would expect to get the raw resource, without it being changed/transpiled, though I'm fine either way to support older systems that might not send the appropriate header.